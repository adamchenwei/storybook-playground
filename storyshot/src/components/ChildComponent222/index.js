import React from 'react'

export default (props) => (<h2>My Custom ChildComponent222 Has Value Of {props.message || 'none'}</h2>)