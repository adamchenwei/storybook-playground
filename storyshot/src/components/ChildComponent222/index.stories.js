import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line
import ChildComponent222 from './index';
import initStoryshots from '@storybook/addon-storyshots';
initStoryshots();

storiesOf('Component/ChildComponent222/Stand Alone', module)
  .add('the default change', () => (
    <ChildComponent222 />
  ))
  .add('another change', () => (
    <ChildComponent222 message={'my thing'} />
  ))
