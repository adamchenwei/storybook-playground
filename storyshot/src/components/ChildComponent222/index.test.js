import React from 'react';
import ChildComponent222 from './index';
import renderer from 'react-test-renderer';
import initStoryshots from '@storybook/addon-storyshots';
initStoryshots();

describe('ChildComponent222 Component', () => {
  test('renders as expected', () => {
    const tree = renderer.create(
      <ChildComponent222 />
    ).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
})