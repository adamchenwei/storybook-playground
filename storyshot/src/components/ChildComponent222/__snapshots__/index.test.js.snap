// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildComponent222 Component renders as expected 1`] = `
<h2>
  My Custom ChildComponent222 Has Value Of 
  none
</h2>
`;

exports[`Storyshots Component/App/Stand Alone another change 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="IMAGE_MOCK"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h2>
      everything111
    </h2>
  </header>
  <p
    className="App-intro"
  >
    To get started22, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`Storyshots Component/App/Stand Alone the default change 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="IMAGE_MOCK"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h2 />
  </header>
  <p
    className="App-intro"
  >
    To get started22, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone another change 1`] = `
<h2>
  My Custom Component Has Value Of 
  my thing
</h2>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone another change 2`] = `
<h2>
  My Custom Component Has Value Of 
  my thing
</h2>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone another change 3`] = `
<h2>
  My Custom Component Has Value Of 
  my thing
</h2>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone the default change 1`] = `
<h2>
  My Custom Component Has Value Of 
  none
</h2>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone the default change 2`] = `
<h2>
  My Custom Component Has Value Of 
  none
</h2>
`;

exports[`Storyshots Component/ChildComponent/Stand Alone the default change 3`] = `
<h2>
  My Custom Component Has Value Of 
  none
</h2>
`;

exports[`Storyshots Component/ChildComponent222/Stand Alone another change 1`] = `
<h2>
  My Custom ChildComponent222 Has Value Of 
  my thing
</h2>
`;

exports[`Storyshots Component/ChildComponent222/Stand Alone another change 2`] = `
<h2>
  My Custom ChildComponent222 Has Value Of 
  my thing
</h2>
`;

exports[`Storyshots Component/ChildComponent222/Stand Alone the default change 1`] = `
<h2>
  My Custom ChildComponent222 Has Value Of 
  none
</h2>
`;

exports[`Storyshots Component/ChildComponent222/Stand Alone the default change 2`] = `
<h2>
  My Custom ChildComponent222 Has Value Of 
  none
</h2>
`;
